src/main/java/ecommerce/dao/impl/jdbc/OrderDaoJdbcImpl.java

```diff
@@ -0,0 +1,182 @@
+package ecommerce.dao.impl.jdbc;
+
+import ecommerce.dao.interfaces.OrderDao;
+import ecommerce.exceptions.DataProcessingException;
+import ecommerce.lib.Dao;
+import ecommerce.model.Order;
+import ecommerce.model.Product;
+import ecommerce.util.ConnectionUtil;
+import java.sql.Connection;
+import java.sql.PreparedStatement;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.sql.Statement;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+
+@Dao
+public class OrderDaoJdbcImpl implements OrderDao {
+    @Override
+    public List<Order> getUserOrders(Long userId) {
+        String query = "SELECT * FROM orders WHERE user_id = ?;";
+        try (Connection connection = ConnectionUtil.getConnection();
+                 PreparedStatement statement
+                         = connection.prepareStatement(query)) {
+            statement.setLong(1, userId);
+            try (ResultSet resultSet = statement.executeQuery()) {
+                List<Order> allOrders = new ArrayList<>();
+                while (resultSet.next()) {
+                    Order order = getOrderFromResultSet(resultSet, connection);
+                    allOrders.add(order);
+                }
+                return allOrders;
+            }
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can't get order of user with id " + userId, e);
+        }
+    }
+
+    @Override
+    public Order create(Order order) {
+        String query = "INSERT INTO orders (user_id) VALUES (?);";
+        try (Connection connection = ConnectionUtil.getConnection();
+                 PreparedStatement statement
+                         = connection.prepareStatement(query,
+                         Statement.RETURN_GENERATED_KEYS)) {
+            statement.setLong(1, order.getUserId());
+            statement.executeUpdate();
+            try (ResultSet resultSet = statement.getGeneratedKeys()) {
+                resultSet.next();
+                order.setId(resultSet.getLong(1));
+                addOrderProducts(order, connection);
```

Comment:
it's better to close statement, because you open a new one in addOrderProducts, same goes for others methods when you pass connection to method

Reply:
I can only close statement in create methods because I cant close them in while loops

Reply:
you should close statement not resultSet